src/main/java/duke/tool/DukeException.java

```diff
@@ -0,0 +1,28 @@
+package duke.tool;
+
+/**
+ * Represents the exception thrown by duke.Duke.
+ */
+public class DukeException extends Exception {
+
+    /**
+     * Constructs a duke.Duke exception.
+     * @param message message of the exception.
+     */
+    public DukeException(String message) {
+        super(message);
+    }
+
+    /**
+     * Outputs exception as a string.
+     * @return string representation of the exception.
+     */
+    @Override
+    public String toString() {
+        String emoji = Emoji.ERROR.toString();
+        String msgForException = "    ____________________________________________________________\n"
+                + "    " + emoji + this.getMessage() + "\n"
+                + "    ____________________________________________________________\n";
+        return msgForException;
+    }
+}
```

Comment:
Perhaps it might be better to abstract out the different types of command methods into different command classes inheriting from a main Command class?